crates/kubelet/src/container/state.rs

```diff
@@ -20,26 +23,41 @@ pub async fn run_to_completion<S: ResourceState<Manifest = Container, Status = S
     initial_state: impl State<S>,
     shared: SharedState<S::SharedState>,
     mut container_state: S,
-    pod: SharedState<Pod>,
+    pod: Manifest<Pod>,
     container_name: ContainerKey,
 ) -> anyhow::Result<()> {
-    let (pod_name, api) = {
-        let initial_pod = pod.read().await.clone();
-        let namespace = initial_pod.namespace().to_string();
-        let name = initial_pod.name().to_string();
-        let api: Api<KubePod> = Api::namespaced(client.clone(), &namespace);
-        (name, api)
-    };
+    let initial_pod = pod.latest();
+    let namespace = initial_pod.namespace().to_string();
+    let pod_name = initial_pod.name().to_string();
+    let api: Api<KubePod> = Api::namespaced(client.clone(), &namespace);
 
     let mut state: Box<dyn State<S>> = Box::new(initial_state);
 
+    // Forward pod updates as container updates.
+    let initial_container = initial_pod.find_container(&container_name).unwrap();
```

Comment:
This `unwrap` should be removed and the error handled here and on line 43

Reply:
Should this error be reported to Kubernetes? If the Container is missing from the Pod manifest, I would expect `patch_container_status` to behave poorly.

Reply:
For line 43, I will log an error, but I dont think it makes sense to stop trying (in case a subsequent Pod does have the container in it).

Reply:
Logging is completely fine for now (as we don't know if the error should be reported to k8s). Definitely don't want a panic though, which the log takes care of